src/Layer/C3DTilesLayer.js

```diff
@@ -3,6 +3,270 @@ import GeometryLayer from 'Layer/GeometryLayer';
 import { init3dTilesLayer, pre3dTilesUpdate, process3dTilesNode } from 'Process/3dTilesProcessing';
 import C3DTileset from 'Core/3DTiles/C3DTileset';
 import C3DTExtensions from 'Core/3DTiles/C3DTExtensions';
+// eslint-disable-next-line no-unused-vars
+import Style from 'Core/Style';
+import { optimizeGeometryGroups } from 'Utils/ThreeUtils';
+
+export class C3DTileFeature {
+    // private fields
+    #layer;
+    #userData;
+    #info;
+    /**
+     * Create a new C3DTileFeature which represent a chunk of the tileContent geometry
+     *
+     * @param {C3DTilesLayer} layer - parent layer
+     * @param {number} tileId - tile id
+     * @param {number} batchId - batch id
+     * @param {Array<{start:number, count:number}>} groups - groups in geometry.attributes matching batchId
+     * @param {object} info - info in the batchTable
+     * @param {object} [userData={}] - some userData
+     */
+    constructor(layer, tileId, batchId, groups, info, userData = {}) {
+        /** @type {C3DTilesLayer} */
+        this.#layer = layer;
+
+        /** @type {number} */
+        this._tileId = tileId;
```

Comment:
Iam using _ to indicate this attribute should only be accessed in this file since C3DTileFeature is going to move to Core/3DTiles it will make no sense Iam going to remove it

Reply:
I move C3DTFeature in another file so I removed it